### DISCUSSION TITLE
Have three questions about model_devi section

### DISCUSSION THREAD

**Discussion by 9Jaden** [deepmodeling/dpgen#824] [Q&A]
1. Knowing that the model_devi job is completed by LAMMPS. I want to know what is the _**pair_style**_ adopted in it? Is it the deep potential gained from the _**init_data**_?
2. I have been studying in surface system and need to fix the bottom layer of the configuration. The _**Selective Dynamics**_ in POSCAR didn't work, so I was wondering how to fix the bottom layer during the model_devi section?
3. Besides, I have got a frozen model from DeePMD-kit, but when I want to do some further simulation job by LAMMPS with it. I always received the error below:

> ERROR on proc 0: Neighbor list overflow, boost neigh_modify one (src/npair_full_bin_atomonly.cpp:89)
> Last command: run             100000
> application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
> [unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
> :
> system msg for write_line failure : Bad file descriptor

And in.lammps:

> units           metal
> boundary        p p p
> atom_style      atomic
> 
> neighbor        1.0 bin
> neigh_modify    every 10 delay 0 check no page 50000 one 5000
> 
> read_data       C24.data
> mass            1 64
> mass            2 12
> 
> pair_style      deepmd graph.pb
> pair_coeff      * *
> 
> velocity        all create 1300.0 23456789
> fix             1 all nvt temp 1300.0 1300.0 0.5
> timestep        0.001
> 
> thermo_style    custom step pe ke etotal temp press vol
> thermo          100
> dump            1 all custom 100 c18.dump id type x y z
> 
> run             100000 

Also data file attached:
[C24data.md](https://github.com/deepmodeling/dpgen/files/9178462/C24data.md)
Want to know how to fix this error. 

Great thanks for any help~

**Comment by HuangJiameng** [comment]
1.
> I want to know what is the pair_style adopted in it

`pair_style` is set as `deepmd` by default. 

>Is it the deep potential gained from the init_data

I do not know what you mean

2. There're many discussions about how to fix layers in LAMMPS's community, such as https://sourceforge.net/p/lammps/mailman/message/32396695/ . Maybe you can search for help there. 

3. The reason for the failure is that there're too many atoms  in one atom's neighbor (just like the log demonstrates).  Maybe you should check your settings for LAMMPS, or check the quality of your Deep Potential Model by doing some validation tests.

**Reply by 9Jaden** [reply]
Thanks. I think i know the answer of question 1

**Comment by taipinghu** [comment]
for your question 2, there are two ways to fix this issue. 
(1) change the code of the dpgen (your-dpgen-path/dpgen/generator/lib/lammps.py) and add the fix butter layer code by setting velocity and force as 0. 
(2) you can prepare a lammps input and set template in your param.json file.

**Reply by 9Jaden** [reply]
Still got problem about changing the code of dpgen. Now I know that I can fix the bottom layer by dividing atoms into diffrent types and add codes about setting velocity and force as 0 for bottom layers' corresponding type. But now the question is that the input---sys_configs of dpgen are POSCAR files, I do not know how to distribute different types to atoms in POSCAR for fixing some of them. I can implement it in lammps's data file but not for POSCAR. Maybe the input sys_configs could be file in .data type?
I dont know if i explain the question well but thanks for any help.

**Reply by taipinghu** [reply]
> 

you mean you want to fix some in POSCAR file? in dpgen iteration, VASP single point calculation will be performed. thus, fix atoms is not meaningful.

**Reply by 9Jaden** [reply]
I want the configurations after model_devi have the bottom layer fixed. And the configurations are derived from the POSCAR I set. Now the question is how to change the code of lammps.py of my dpgen to only fix the bottom layer, since the POSCAR type of file can not lable out the bottom layer atoms.

**Reply by taipinghu** [reply]
> I want the configurations after model_devi have the bottom layer fixed. And the configurations are derived from the POSCAR I set. Now the question is how to change the code of lammps.py of my dpgen to only fix the bottom layer, since the POSCAR type of file can not lable out the bottom layer atoms.

fix atom according to the atomic index since the atomic ids cannot be changed during the whole process.

**Reply by 9Jaden** [reply]
but for different POSCARs I used, the number of total atoms changes so that the bottom layers' indexes change too.

**Reply by taipinghu** [reply]
> but for different POSCARs I used, the number of total atoms changes so that the bottom layers' indexes change too.

OK, I see.  You should change the code of the dpgen. you can insert some code in dpgenpath/generator/run.py. when the make_model_devi function finish, replace fix group atomic ids in in.lammps file.
